airflow/providers/databricks/sensors/databricks.py

```diff
@@ -0,0 +1,231 @@
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+"""This module contains Databricks sensors."""
+
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+from typing import Any, Callable, Sequence
+
+from airflow.exceptions import AirflowException
+from airflow.providers.common.sql.hooks.sql import fetch_all_handler
+from airflow.providers.databricks.hooks.databricks_sql import DatabricksSqlHook
+from airflow.sensors.base import BaseSensorOperator
+from airflow.utils.context import Context
+
+
+class DatabricksSqlSensor(BaseSensorOperator):
+    """
+    Generic Databricks SQL sensor.
+
+    :param databricks_conn_id:str=DatabricksSqlHook.default_conn_name: Specify the name of the connection
+    to use with Databricks on Airflow
+    :param http_path:str: Specify the path to the sql endpoint
+    :param sql_endpoint_name:str: Specify the name of the sql endpoint to use
+    :param session_configuration: Pass in the session configuration to be used
+    :param http_headers:list[tuple[str, str]]: Pass http headers to the databricks API
+    :param catalog:str|None=None: Specify the catalog to use for the query
+    :param schema:str|None="default": Specify the schema of the table to be queried
+    :param table_name:str: Specify the table that we want to monitor
+    :param partition_name:dict|: Pass in the partition name to be used for the sensor
+    :param handler:Callable[[Any, Any]=fetch_all_handler: Define the handler function that will be used
+    to process the results of a query
+    :param db_sensor_type:str: Choose the sensor you want to use. Available options: table_partition,
+    table_changes.
+    :param timestamp:datetime: To be used with query filters or as other argument values for timestamp
+    :param caller: Identify the source of this sensor in logs
+    :param client_parameters:dict[str, Any]: Additional client parameters
+    """
+
+    def __init__(
+        self,
+        *,
+        databricks_conn_id: str = DatabricksSqlHook.default_conn_name,
+        http_path: str | None = None,
+        sql_endpoint_name: str | None = None,
+        session_configuration=None,
+        http_headers: list[tuple[str, str]] | None = None,
+        catalog: str = "",
+        schema: str | None = "default",
+        table_name: str,
+        partition_name: dict = {"date": "2023-1-1"},
+        handler: Callable[[Any], Any] = fetch_all_handler,
```

Comment:
Usually for sensor it's enough to just to understand if we have any result or not - we may not need to fetch all results